### PR TITLE
Fix/tests/cleanup

### PR DESCRIPTION
**Pull request by aensidhe**
- Fix tests to be deterministic
- Fix error in EVAL semantic
- Fix error in CALL semantic
- Implement CALL 1.7 semantic
- Handle nulls in response

### CODE REVIEW

File: src/tarantool.client/Model/TarantoolTuple.cs [hunk 1]
```diff
@@ -30,26 +31,35 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            var hashCode = EqualityComparer<T1>.Default.GetHashCode(Item1);
-            return hashCode;
+            unchecked
```
Comment (roman-kozachenko) [inline src/tarantool.client/Model/TarantoolTuple.cs]: Do we really need unchecked{} section here?
What about other TarantoolTuple classes (<T1,T2> and so on)?

File: src/tarantool.client/NetworkStreamPhysicalConnection.cs [hunk 2]
```diff
@@ -30,7 +31,7 @@ public void Dispose()
 
             _socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
 #if PROGAUDI_NETCORE
```
Comment (roman-kozachenko) [inline src/tarantool.client/NetworkStreamPhysicalConnection.cs]: Looks like you don't need that #if anymore.

File: tarantool/tarantool.lua [hunk 3]
```diff
@@ -21,33 +21,59 @@ local function init()
...
-    box.session.on_disconnect(log_disconnect)
-    box.session.on_auth(log_auth)
-    box.session.on_auth(log_auth_ok)
-end
-
-box.once('init', init)
\ No newline at end of file
+function return_scalar()
+    return 1
+end
```
Comment (roman-kozachenko) [inline tarantool/tarantool.lua]: Did you test what happens when you call a function without return statement?

File: src/tarantool.client/ResponseReader.cs [hunk 4]
```diff
@@ -139,6 +140,27 @@ private void MatchResult(byte[] result)
             var header= MsgPackSerializer.Deserialize<ResponseHeader>(resultStream, _clientOptions.MsgPackContext);
             var tcs = _logicalConnection.PopResponseCompletionSource(header.RequestId, resultStream);
 
+            if (tcs == null)
+            {
+                if (_clientOptions.LogWriter == null)
+                    return;
+
+                var builder = new StringBuilder("Warning: can't match request via requestId from response. Response:");
```
Comment (roman-kozachenko) [inline src/tarantool.client/ResponseReader.cs]: Extract to separate method

File: src/tarantool.client/Model/TarantoolTuple.cs (hunk 1 above)
Comment (aensidhe) [inline src/tarantool.client/Model/TarantoolTuple.cs]: @roman-kozachenko in this class - no we do not need. But we need it in other classes. So, as this code is autogenerated, I will not edit it.

File: src/tarantool.client/ResponseReader.cs (hunk 4 above)
Comment (aensidhe) [inline src/tarantool.client/ResponseReader.cs]: Done

File: tarantool/tarantool.lua (hunk 3 above)
Comment (aensidhe) [inline tarantool/tarantool.lua]: @roman-kozachenko ok. I added tests and separate overload for call for this

File: src/tarantool.client/Model/TarantoolTuple.cs (hunk 1 above)
Comment (roman-kozachenko) [inline src/tarantool.client/Model/TarantoolTuple.cs]: @aensidhe Actually I already found that, forgot to remove comment

**Pull request merged**